core/store/src/trie/trie_storage.rs

```diff
@@ -16,7 +18,23 @@ pub struct TrieCache(Arc<Mutex<LruCache<CryptoHash, Arc<[u8]>>>>);
 
 impl TrieCache {
     pub fn new() -> Self {
-        Self(Arc::new(Mutex::new(LruCache::new(TRIE_MAX_CACHE_SIZE))))
+        Self::with_capacity(TRIE_MAX_SHARD_CACHE_SIZE)
```

Comment:
I'd mention above, that TrieCache is a wrapper over the LruCache, but doesn't hold too large elements